skbio/stats/power.py

```diff
@@ -1072,8 +1072,14 @@ def _draw_paired_samples(meta_pairs, index, num_samps):
 
     subs = []
 
-    # Draws the other groups
-    for set_, num_ in collections.Counter(set_pos).items():
+    # Draws the other groups. Get a collection.Counter object for simplicity
+    counter = collections.Counter(set_pos)
+
+    # counter.items() order isn't guaranteed in python 3.6 and then the random
+    # coiche isn't reproducible between python version, even specifying seed
+    # so we access such values through sets
```

Comment:
Ok, I added an intermediate step to ensure that the list derived from set is ordered